tests/core/test_oauth_clients.py

```diff
@@ -0,0 +1,53 @@
+from flask import json
+from newsroom.tests.users import (test_login_succeeds_for_admin, init as user_init)  # noqa
+from superdesk import get_resource_service
+from newsroom.auth_server.auth import JWTAuth
+import base64
+
+
+def test_oauth_clients(client):
+    test_login_succeeds_for_admin(client)
+    # Register a new client
+    response = client.post(
+        "/oauth_clients/new",
+        data=json.dumps({"name": "client11"}),
+        content_type="application/json",
+    )
+    assert response.status_code == 201
+
+    # Check for the client secret
+    password = response.json.get("password", None)
+    if not password:
+        assert False
+
+    # OAuth Token Generation
+    username = response.json["_id"]
+    userpass = username + ":" + password
+    encoded_u = base64.b64encode(userpass.encode()).decode()
+
+    payload = {"grant_type": "client_credentials"}
+    token_auth_response = client.post(
+        "api/auth_server/token",
+        headers={"Authorization": "Basic %s" % encoded_u},
+        data=payload,
+    )
+    token = token_auth_response.json["access_token"]
```

Comment:
you can check the response code on token_auth_response, that should give some hint what is wrong on the server.
and you could also add the client_id and client_secret to the payload, maybe it's an issue with the basic auth

Reply:
While running locally, I am able to get `access_token` from `token_auth_response`.
And I am using `client_id` and `client_secret` to generate `encoded_u` which is used with `Authorization` in `headers`.
And It works fine locally. Just little doubt full about it running on server.

Reply:
I see, so what do you suggest?